hledger-lib/Hledger/Data/Journal.hs

```diff
@@ -463,8 +470,8 @@ journalApplyAliases aliases j@Journal{jtxns=ts} =
 -- check balance assertions.
 journalFinalise :: ClockTime -> FilePath -> Text -> Bool -> ParsedJournal -> Either String Journal
 journalFinalise t path txt assrt j@Journal{jfiles=fs} = do
-  (journalNumberAndTieTransactions <$>
-    (journalBalanceTransactions $
+  (journalTieTransactions <$>
```

Comment:
The numbering now is performed in  `journalBalanceTransactions`, which uses it to get back the parsing order after iterating over time ordered postings.